### PR TITLE
adding the ability to send to irc channels with passwords

### PR DESCRIPTION
**Pull request by lum**
this will work with the changes in irc-chef-snitch to use irc channel passwords when joining channels and sending messages.

### CODE REVIEW

**Comment by portertech** [comment]
maybe channel could be either a string "#pigeon", or a ruby hash {:name => "#pigeon", :password => "channel_password"}? or use an option like option[:channel_password]?

**Comment by lum** [comment]
Ok, i've added the option{:channel_password]

So you can run it like:

require 'rubygems'
require 'carrier-pigeon'

 CarrierPigeon.send(:uri => "irc://irc_nick:password@irc.freenode.net:6667/#secret_chan", :message => "yo yo yo", :channel_password => "passwordz", :join => true)

as well as using it from chef-irc-snitch

File: lib/carrier-pigeon.rb [hunk 1]
```diff
@@ -43,6 +43,7 @@ class CarrierPigeon
     options[:port] = uri.port || 6667
     options[:nick] = uri.user
     options[:password] = uri.password
+    options[:channel_password] = options[:channel_password]
```
Comment (portertech) [inline lib/carrier-pigeon.rb]: don't need this line, as it passes through w/ new(options)

**Comment by lum** [comment]
ok that line is removed.

**Comment by portertech** [comment]
Thanks guys! A new version (0.3.0) of Carrier-Pigeon thanks to you https://rubygems.org/gems/carrier-pigeon

**Pull request merged**